usr/src/cmd/zfs/zfs_iter.c

```diff
@@ -391,7 +392,7 @@ zfs_for_each(int argc, char **argv, int flags, zfs_type_t types,
 		}
 
 		while (sortcol) {
-			if (sortcol->sc_prop >= ZFS_PROP_TYPE &&
+			if ((int)sortcol->sc_prop >= ZFS_PROP_TYPE &&
```

Comment:
only gcc44 complain here with 
zfs_iter.c:394: error: comparison of unsigned expression >= 0 is always true [-Wtype-limits]

and gcc48 work without warning here.
what you are prefer - put back line to Makefile:
CERRWARN += -_gcc=-Wno-type-limits

for hide this warning?